stdlib/test/builtin/test_string.mojo

```diff
@@ -828,6 +828,17 @@ def test_string_mul():
     assert_equal(String("ab") * 5, "ababababab")
 
 
+def test_empty_string():
+    assert_equal(str(String()), String())
+    assert_equal(str(String()), String(""))
+    assert_equal(str(String()), str(String("")))
+    assert_equal(str(String()), "")
+    assert_equal(str(String()), str(""))
+    assert_equal(len(String()), 0)
+
+    assert_equal(String(), String(List[Int8]()))
+
+
```

Comment:
One of the main benefits I can see from this change is that an empty `String` is now validly null terminator, so it can be passed to functions that are expecting a non NUL string pointer.

So if thats the desired behavior, we should add some tests for that🙂 

But, I'd argue most functions that could accept an empty string should arguably be written to accept a NUL pointer, because as mentioned then they'll optimally allow the caller to avoid allocating space just to store an empty string.1

Reply:
>So if thats the desired behavior, we should add some tests for that🙂

Do you mean adding a test where if do `assert_equal(String()._buffer[0], 0)`?

Reply:
Done :)